velox/exec/HashTable.cpp

```diff
@@ -669,9 +671,8 @@ void HashTable<ignoreNullKeys>::checkSize(int32_t numNew) {
     // then the table lookup will become slow. Given that, we treat tombstone
     // slot as non-empty slot here to decide whether to trigger rehash or not.
   } else if (newNumDistincts > rehashSize()) {
-    // NOTE: we need to plus one here as newNumDistincts itself could be power
-    // of two.
-    auto newSize = bits::nextPowerOfTwo(newNumDistincts + 1);
+    auto newSize = bits::nextPowerOfTwo(stats_.capacity + 1);
```

Comment:
Good catch.
```
bits::nextPowerOfTwo(size_ - numTombstones_ + 1);
```

Reply:
@xiaoxmeng 
If we rehash() would tombstones disappear and essentially turn into the empty slots?
Would we run into more inefficiency with not bumping up the capacity due to tombstones due to calling rehash() more often than we want it, since rehash() is expensive)?
Would it make sense to gauge the magnitude of numTombstones_ instead of blindly subtract that?

Say having capacity_ = 1024 and numTombstones_ = 2 we still want to allocate to the next power of two of the capacity: capacity_ -> 2048, but with you suggestion we'll stay with 1024 and would need to rehash() again. That loop can repeat.
But if we have capacity_ = 1024 and numTombstones_ = 200, maybe we don't want yet to got to 2048...

Should we take the same 1/8 threshold that rehashSize() method is using or something else?